src/fu-device-list.c

```diff
@@ -580,11 +580,14 @@ fu_device_list_replace (FuDeviceList *self, FuDeviceItem *item, FuDevice *device
 	fu_device_list_add_missing_guids (device, item->device);
 
 	/* enforce the vendor ID if specified */
-	if (fu_device_get_vendor_id (item->device) != NULL &&
-	    fu_device_get_vendor_id (device) == NULL) {
-		const gchar *vendor_id = fu_device_get_vendor_id (item->device);
-		g_debug ("copying old vendor ID %s to new device", vendor_id);
-		fu_device_set_vendor_id (device, vendor_id);
+	if (fu_device_get_vendor_ids (item->device)->len > 0 &&
+	    fu_device_get_vendor_ids(device)->len == 0) {
+		GPtrArray *vendor_ids = fu_device_get_vendor_ids (item->device);
+		for (guint i = 0; i < vendor_ids->len; i++) {
+			const gchar *vendor_id = g_ptr_array_index (vendor_ids, i);
+			g_debug ("copying old vendor ID %s to new device", vendor_id);
+			fu_device_add_vendor_id (device, vendor_id);
+		}
```

Comment:
just wondering if there is possibly a bug here where there are two vendor IDs on donor and one on the new.  possibly these are the same and you just need the one from the donor?

Reply:
if they're the same they get deduped by fu_device_add_vendor_id -- or do I misunderstand?

Reply:
But you never run the copy the ID code path above if the new device has a single ID set.  Right now it's a conservative, "don't destroy any IDs if something is set", "don't add IDs if something is set".  So I think it should be either replace all ID's or add to the existing IDs if some are set.

Reply:
Ohh I see what you mean, gotcha. I'm tempted to say "always copy vendor IDs" is probably the sanest behaviour; it's the same device afterall... i.e. we'd remove both conditionals in the `if` section.

Reply:
Sounds good.

Reply:
Fixed.